src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -502,7 +529,17 @@ internal void Grow(int capacity)
             // If the computed capacity is still less than specified, set to the original argument.
```

Comment:
Requested changes:

1. Document the algorithm for initialization of `newCapacity`
2. For initial length greater than or equal to half the segment size but less than one full segment size, set `newCapacity` to one segment size.
3. For any initial length where the final segment is not a full segment, set `newCapacity` to the length it would be with a full-size final segment. This guarantees that the outer array will not be reallocated, and also guarantees that the single inner array allocation performed during the resize will not need to be performed a second time during the next resize. (Can be modified and treated as a generalization of the preceding point)
4. If the calculated `newCapacity` ends up being less than `capacity` _and_ `capacity` is greater than the segment size, apply a final ceiling operation so the final segment is full size.

These rules are all relevant regardless of whether we increase by doubling or by a page at a time. I am still reviewing the choice of expansion size.

Reply:
➡️ Changing the capacity selection algorithm for resize has been deferred to a later pull request.